pkg/app/piped/cmd/launcher/launcher.go

```
@@ -0,0 +1,413 @@
+		return nil, fmt.Errorf("failed to download Piped from %s to %s (%w)", downloadURL, binaryDir, err)
+	}
+	logger.Info(fmt.Sprintf("LAUNCHER: downloaded Piped binary from %s to %s", downloadURL, pipedPath))
+
+	// Write Piped configuration into working directory.
+	var (
+		configFileDir  = filepath.Join(workingDir, "config")
+		configFilePath = filepath.Join(configFileDir, pipedConfigFileName)
+	)
+	if err := os.MkdirAll(configFileDir, 0755); err != nil {
+		return nil, fmt.Errorf("failed to create directory %s (%w)", configFileDir, err)
+	}
+	if err := os.WriteFile(configFilePath, config, 0644); err != nil {
+		return nil, fmt.Errorf("failed to write Piped config data to file at %s (%w)", configFilePath, err)
+	}
+	logger.Info(fmt.Sprintf("LAUNCHER: wrote Piped config to %s", configFilePath))
+
+	args := makePipedArgs(os.Args[2:], configFilePath)
+	logger.Info(fmt.Sprintf("LAUNCHER: start running Piped %s with args %v", version, args))
+
+	return runBinary(pipedPath, args)
+}
+
+func (l *launcher) loadConfigData(ctx context.Context) ([]byte, error) {
+	// Load config data from the specified file.
+	if l.configFile != "" {
+		return os.ReadFile(l.configFile)
+	}
+
+	// Return config data passed directly.
+	if l.configData != "" {
+		return []byte(l.configData), nil
+	}
+
+	// Load config data from a secret which is stored in Google Cloud Secret Manager service.
+	if l.configFromGCPSecret {
+		if l.gcpSecretID == "" {
+			return nil, fmt.Errorf("gcp-secret-id is required to load Piped config from GCP SecretManager service")
+		}
+
+		client, err := secretmanager.NewClient(ctx)
+		if err != nil {
+			return nil, err
+		}
+		defer client.Close()
+
+		req := &secretmanagerpb.AccessSecretVersionRequest{
+			Name: l.gcpSecretID,
+		}
+		resp, err := client.AccessSecretVersion(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		return resp.Payload.Data, nil
+	}
+
+	if l.configFromGitRepo {
+		// TODO: Support loading config data from a Git repository.
```

nits, should return an error unsupported/not yet supported configuration.